[Bug]: "Cannot find module './index' or its corresponding type declarations" in react-router-dom/server.d.ts

### What version of React Router are you using?

6.23.0

### Steps to Reproduce

Install react-router-dom in a TS project, and attempt to compile the project.

### Expected Behavior

The project compiles.

### Actual Behavior

An error is raised during compilation: 

```
node_modules/react-router-dom/server.d.ts:3:58 - error TS2307: Cannot find module './index' or its corresponding type declarations.

3 import type { FutureConfig, Location, RouteObject } from "./index";
                                                           ~~~~~~~~~
```

--- a/.changeset/little-ghosts-give.md
+++ b/.changeset/little-ghosts-give.md
@@ -1,0 +1,5 @@
+---
+"react-router-dom": patch
+---
+
+Change the `react-router-dom/server` import back to `react-router-dom` instead of `index.ts`

--- a/packages/react-router-dom/server.tsx
+++ b/packages/react-router-dom/server.tsx
@@ -29,7 +29,7 @@ import type {
   Location,
   RouteObject,
   To,
-} from "./index";
+} from "react-router-dom";
 import {
   createPath,
   parsePath,
@@ -38,7 +38,7 @@ import {
   UNSAFE_DataRouterStateContext as DataRouterStateContext,
   UNSAFE_FetchersContext as FetchersContext,
   UNSAFE_ViewTransitionContext as ViewTransitionContext,
-} from "./index";
+} from "react-router-dom";
 
 export interface StaticRouterProps {
   basename?: string;

--- a/packages/react-router-dom/tsconfig.json
+++ b/packages/react-router-dom/tsconfig.json
@@ -16,6 +16,12 @@
     "skipLibCheck": true,
 
     "outDir": ".",
-    "rootDir": "."
+    "rootDir": ".",
+
+    // Needed for the more-strict pnpm setup, can be removed in v7 since we'll
+    // be collapsing the lifted server file into the main package export
+    "paths": {
+      "react-router-dom": ["./index.tsx"]
+    }
   }
 }

